src/ast/passes/codegen_llvm.cpp

```diff
@@ -2430,9 +2430,13 @@ void CodegenLLVM::visit(AssignMapStatement &assignment)
 void CodegenLLVM::maybeAllocVariable(const std::string &var_ident,
                                      const SizedType &var_type)
 {
-  if (variables_.find(var_ident) != variables_.end()) {
-    return;
+  for (auto scope : scope_stack_) {
+    if (auto search_val = variables_[scope].find(var_ident);
+        search_val != variables_[scope].end()) {
```

Comment:
If `getVariable` returned a pointer (or `nullptr` if the variable doesn't exist), you could also use it here.

Reply:
I thought about that but I don't want to be sprinkling checks and error message logic all over codegen. I could add a `hasVariable` function that `getVariable` could call.

Reply:
Right, yeah, that would make sense.